tests/Console/ConfigurationResolverTest.php

```diff
@@ -868,6 +869,123 @@ public function testResolveRulesWithOption()
         );
     }
 
+    /**
+     * @param string[] $rules
+     * @param string[] $negativeRules
+     *
+     * @dataProvider provideSubtractionCases
+     */
+    public function testResolveRulesWithNegativeSet(array $rules, array $negativeRules)
```

Comment:
Please redesign the test, now it's messy.
What is the expected value here? nobody knows.

Input should be `string $rules, array $expectedRules`